hydra-main/src/main/java/com/addthis/hydra/job/alias/AliasManagerImpl.java

```diff
@@ -51,14 +99,158 @@ public AliasManagerImpl(SpawnDataStore spawnDataStore) {
      */
     public void addAlias(String alias, List<String> jobs) {
         if (jobs.size() > 0) {
-            aliasBiMap.putAlias(alias, jobs);
+            putAlias(alias, jobs);
         } else {
             log.warn("Ignoring empty jobs addition for alias: {}", alias);
         }
     }
 
+    public void putAlias(String alias, List<String> jobs) {
+        mapLock.lock();
+        try {
+            alias2jobs.put(alias, jobs);
+            job2alias.put(jobs.get(0), alias);
+            StringWriter sw = new StringWriter();
+            mapper.writeValue(sw, jobs);
+            spawnDataStore.putAsChild(ALIAS_PATH, alias, sw.toString());
+        } catch (Exception e) {
+            log.warn("failed to put alias: {}", alias, e);
+            throw Throwables.propagate(e);
+        } finally {
+            mapLock.unlock();
+        }
+    }
+
+    /**
+     * Get all jobIds for a given alias
+     *
+     * @param alias The alias to check
+     * @return A list of jobIds, possible null
+     */
+    public List<String> getJobs(String alias) {
+        refreshAlias(alias);
+        mapLock.lock();
+        try {
+            return alias2jobs.get(alias);
+        } finally {
+            mapLock.unlock();
+        }
+    }
+
+    /**
+     * Refresh an alias based on datastore
+     *
+     * @param alias The alias to refresh
+     */
+    private void refreshAlias(String alias) {
+        try {
+            updateAlias(alias, this.spawnDataStore.getChild(ALIAS_PATH, alias));
+        } catch (ExecutionException e) {
+            log.warn("Failed to refresh alias: {}", alias, e);
+        } catch (Exception e) {
+            log.error("",e);
+        }
+    }
+
+    /**
+     * Load the jobIds for a particular alias from the SpawnDataStore
+     *
+     * @param alias The alias key to check
+     * @return String The data that was updated (so the cache can be updated)
+     */
+    @Nullable private String updateAlias(String alias, @Nullable String data) {
```

Comment:
This method has unclear side effects IMO. It seems to do an update or delete based on whether or not the data argument is passed. The java docs need to be updated.